### DISCUSSION TITLE
CrazyFile bolt disconnect

### DISCUSSION THREAD

**Discussion by maichongju** [bitcraze/discussions#282] [Q&A]
this problem happens from time to time and I cannot reproduce it. After the bolt takes off, sometimes it will just stop and drop from the air. It did not call back any disconnect function from the `cflib`. From the bolt itself, I can see the connection light is no longer visible. But the red light is flashing twice a second which should indicate is ready to fly. But then if I try to connect the bolt again, it won't connect to it. It will just be stuck at `request connection setup` unless I restart the bolt.

Because there isn't any feedback from the drone about why did it disconnect. Therefore it is hard for me to find out what is causing this. 

Just wondering if anyone has experienced this issue before? If so, how should I find out what is causing the disconnection?

**Comment by krichardsson** [comment]
Which version of the firmware are you using?

**Reply by maichongju** [reply]
The firmware was a custom compile version of the 2022.05. We only change the PID value and some other settings. None logic change or update. 
I just saw that there is a new firmware 2022.09. Perhaps I should try the new firmware and see if the issue still exist.

**Comment by krichardsson** [comment]
It sounds like your NRF might have crashed, but I'm not sure. 

We are currently investigating intermittent problems on the syslink. Syslink is the link between the NRF and STM CPUs and we have seen issues such as lost packets and that one of the CPUs might halt. Versions 2022.05 and 2022.09 are pretty stable as far as we know and the problems have mainly been seen in versions in-between. As you used 2022.05 it should be very unlikely.

How often does it happen to you?
What does your set up looks like, decks and so on?

The NRF handles radio communication and forwards the packets to the STM via syslink. USB on the other hand is connected directly to the STM and USB communication does not use syslink. 

To investigate this further I hope you could make a test when you are in this state next time. I would like you to try to connect to the Crazyflie via USB instead of the radio. It a USB connection works it would indicate that the STM still is OK, but the NRF is dead or syslink is not working.

You could try to 2022.09 as well. We have made a small change related to syslink that possibly could make a difference.

**Comment by maichongju** [comment]
I have built the firmware with the latest 2022.09 but the issue still exists. 

the issue happens from time to time. Sometimes 1 in every 10 flights. But sometimes more often. 

When the drone is in that state, I plug the drone via USB and the PC can recognize the drone. So I believe then the STM is ok.

**Comment by krichardsson** [comment]
This sounds like it could be the NRF, we have not seen it that much in 2022.09 though.
We need more information about your set up to be able to help you further.

**Comment by maichongju** [comment]
Sure. What kind of information do you need? 

We are running the bolt with three decks, flow deck, loco deck and the multi-ranger. 

Software: cflib-0.1.20.1, python 3.10

**Reply by tobbeanton** [reply]
I tried replicating it but couldn't trigger the problem. I didn't fly though. What I did was to connect all decks, flash with 2022.09, connect it to cfclient while in the LPS system and waited for about an hour.

Can you trigger it without flying? All details that could help replicating the problem are valuable.

**Reply by maichongju** [reply]
This is the part that is going to be hard. The problem only happens from time to time. And I am not able to replicate the issue consistently. 

Previously, it always happen after take-off. But yesterday it went into the same situation right after I connect it via the cflib (Have not even taken off). After restart, everything went back to normal. 

Is there any measurement that I can add to the crazyfile so that it can always capture some information that can help you to find out the issue and I can share that information once the situation appears again?

**Reply by krichardsson** [reply]
Sorry for the late reply, we're at IROS and support is/will be slower for some days.

I was wondering if this problem can be related to communication in any way. How much communication is going on in your use case? Have you set up a lot of logging or are you sending a lot of data to the Crazyflie? 

> Is there any measurement that I can add to the crazyfile so that it can always capture some information that can help you to find out the issue and I can share that information once the situation appears again?

Unfortunately there is currently no good way of capturing problems in the NRF

**Reply by maichongju** [reply]
I do have quite a bit of log register to it. that might be the issue. So here is the setup I have. 

1. Battery Logger ( BatteryLevel, BatteryVoltage,BatteryState), update every 100ms
2. Core logger ( Pitch, roll, yaw)update every 10ms
3. MotorThrustLogger(Core_thrust,M1,M2,M3,M4) update every 250ms
4. multiRangerLogger(Front,left,right,rear,down) update every 10ms
5. PositionLogger(stateEstimate.x,stateEstimate.y,stateEstimate.z) update every 10ms
6. LocoLogger(locoMode), update every 100ms

My client is sending commands to the drone around 9 times per second.

**Reply by tobbeanton** [reply]
Sorry for a late reply and it's been to busy to try to replicate the problem. Hopefully we will find some time soon.

**Reply by wdliu356** [reply]
Hi, is there any update here? I think I'm facing the same problem but using bigquad deck though. Is it the large amount of logging that caused such problem?

**Reply by tobbeanton** [reply]
Have you managed to replicate the problem on a normal Crazyfie?

**Reply by wdliu356** [reply]
I'm now working on large quad so maybe I can try with the bolt first. Finally got my bolt delivered. But based on my experiment before, I didn't meet this problem when working with normal crazyflie.
